### ISSUE TITLE
Enhancement: MicroProfile Config implementation should leverage getPropetyNames() for better performance

### ISSUE THREAD

**Issue by ghunteranderson** [payara/Payara#5187]
# Description #
----------
In v1.1, MicroProfile Config added default method `Set<String> getPropertyNames()` to `ConfigSource`.  If a config source is not very performant, then the source could override `getPropertyNames()` and provide a faster implementation of retrieving only the config names. However, Payara's [implementation](https://github.com/payara/Payara/blob/master/nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java#L192) of `Config` does not use this method. Instead it loads all configuration names and evaluates the values before discarding the values.

## Expected Outcome

When getting an Iterable of property names, I would expect PayaraConfig to use `ConfigSource.getPropertyNames()` giving `ConfigSource` implementers the option to provider a faster method for getting names. A stack trace might look like this:
```
at com.ghunteranderson.CustomConfigSource.getPropertyNames(CustomConfigSource.java)
at fish.payara.nucleus.microprofile.config.spi.PayaraConfig.getPropertyNames(PayaraConfig.java:105)
at fish.payara.microprofile.openapi.impl.config.OpenApiConfiguration.findPathServerMap(OpenApiConfiguration.java:301)
```

## Current Outcome

Currently, PayaraConfig loads all properties (name and value) before discarding the value. It does not give implementers an option to provider faster name-only lookup method that MicroProfile intended when adding `ConfigSource.getPropertyNames()`. A stack trace might look like this:

```
at com.ghunteranderson.CustomConfigSource.slowLookupOperation(CustomConfigSource.java)
at com.ghunteranderson.CustomConfigSource.getProperties(CustomConfigSource.java)
at fish.payara.nucleus.microprofile.config.spi.PayaraConfig.getPropertyNames(PayaraConfig.java:105)
at fish.payara.microprofile.openapi.impl.config.OpenApiConfiguration.findPathServerMap(OpenApiConfiguration.java:301)
```

## Context
In my use case, I've added a config source that makes network calls to load configuration. It's much faster for me to get a list of config names than to load every config value.

I believe I can make a PR for this change but first I wanted to get some feedback on if this is something that should be changed. Thanks!

**Comment by smillidge** [comment]
Feel free to raise a PR.

**Comment by AlanRoth** [comment]
Hi @ghunteranderson, thanks for offering to raise a PR - feel free to do so.

I have put your issue into voting status to perhaps garner some community feedback, and if you desire you can leave it in voting status and if the issue is passed with enough votes - we will accept and implement the feature.

Thank you,
Alan

**Comment by smillidge** [comment]
Fixed by #5200
